static/src/deploys-radiator/app/api.ts

```diff
@@ -43,15 +43,15 @@ export const getDifference = (base: string, head: string): Promise<List<GitHubCo
                         {
                             url: gitHubCommitJson.html_url,
                             authorName: gitHubCommitJson.commit.author.name,
-                            authorLogin: gitHubCommitJson.author.login,
+                            authorLogin: (gitHubCommitJson.author) ? gitHubCommitJson.author.login : 'Unknown',
```

Comment:
This error slipped through the static type system because, currently in TypeScript, each type can be itself or `undefined`, e.g. `Author = Author | undefined`. This means you sometimes see `foo is not a {property,method} of undefined` errors slipping through the net. TypeScript doesn't know whether `gitHubCommitJson.author` is `Author` or `undefined`.

TypeScript 2 has a feature called control-flow based analysis which fixes this. It will follow the control flow of your program to determine the type. The above (`gitHubCommitJson.author.login`) would throw a compile error, but `gitHubCommitJson.author && gitHubCommitJson.author.login` would not. It forces you to guard these type of errors.

Another cool feature of TypeScript 2 is non-nullable types, which means `Author = Author`, instead of `Author = Author | undefined`.